rotkehlchen/chain/evm/decoding/curve/lend/common.py

```diff
@@ -0,0 +1,330 @@
+
+    def _maybe_get_cached_address_from_contract(
+            self,
+            cache_type: Literal[
+                CacheType.CURVE_LENDING_VAULT_CONTROLLER,
+                CacheType.CURVE_LENDING_VAULT_BORROWED_TOKEN,
+                CacheType.CURVE_CRVUSD_COLLATERAL_TOKEN,
+                CacheType.CURVE_CRVUSD_AMM,
+            ],
+            contract_address: 'ChecksumEvmAddress',
+            contract_abi: ABI,
+            contract_method: Literal['amm', 'controller', 'collateral_token', 'borrowed_token'],
+    ) -> ChecksumEvmAddress | None:
+        """Get address from the cache or the specified vault contract method.
+        Returns the address or None on error."""
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            if (value := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(cache_type, str(contract_address)),
+            )) is not None:
+                return string_to_evm_address(value)
+
+        try:
+            if not (value := self.evm_inquirer.call_contract(
+                contract_address=contract_address,
+                abi=contract_abi,
+                method_name=contract_method,
+            )):
+                log.error(f'Empty response from {contract_method} method on Curve contract {contract_address}.')  # noqa: E501
+                return None
+
+            # Use to_checksum_address to fix lower case addresses
+            value = to_checksum_address(value)
+        except RemoteError as e:
+            log.error(f'Failed to call the {contract_method} method on Curve contract {contract_address} due to {e!s}')  # noqa: E501
+            return None
+
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(cache_type, str(contract_address)),
+                value=value,
+            )
+        return value
+
+    def _maybe_get_cached_token(
+            self,
+            cache_type: Literal[
+                CacheType.CURVE_LENDING_VAULT_BORROWED_TOKEN,
+                CacheType.CURVE_CRVUSD_COLLATERAL_TOKEN,
+            ],
+            contract_address: 'ChecksumEvmAddress',
+    ) -> 'EvmToken | None':
+        """Get token from the cache or the specified vault contract method.
+        Returns the token or None on error."""
+        if cache_type == CacheType.CURVE_CRVUSD_COLLATERAL_TOKEN:
+            contract_method = 'collateral_token'
+            contract_abi = CURVE_VAULT_ABI
+        else:
+            contract_method = 'borrowed_token'
+            contract_abi = CURVE_VAULT_ABI
+
+        if (token_address := self._maybe_get_cached_address_from_contract(
+            cache_type=cache_type,
+            contract_address=contract_address,
+            contract_abi=contract_abi,
+            contract_method=contract_method,  # type: ignore  # hardcoded to the correct methods above
```

Comment:
you can just define the variable instead of a type ignore.
`contract_method: Literal['method1', 'method2']`